setup.py

```diff
@@ -134,50 +157,60 @@ def ssl_context():
     return ctx
 
 
-DECAY_H = os.path.join('src', 'decay.h')
-DECAY_CPP = os.path.join('src', 'decay.cpp')
-DECAY_H_REP = os.path.join('src', '_decay.h')
-DECAY_CPP_REP = os.path.join('src', '_decay.cpp')
-DECAY_URL = 'http://raw.githubusercontent.com/pyne/data/master/decay.tar.gz'
-
-
-def download_decay():
-    print('Downloading ' + DECAY_URL)
-    durl = urlopen(DECAY_URL, context=ssl_context())
+def download_and_extract(url, target_dir='.'):
+    print('Downloading ' + url)
+    durl = urlopen(url, context=ssl_context())
     try:
         d = durl.read()
     except IOError:
-        print('...failed to download!')
+        print('...failed to download!', file=sys.stderr)
         return False
     finally:
         durl.close()
     f = io.BytesIO(d)
-    tar = tarfile.open(fileobj=f, mode='r:gz')
-    tar.extractall('src')
+    try:
+        tar = tarfile.open(fileobj=f, mode='r:gz')
+    except tarfile.ReadError:
+        print('...failed to extract!', file=sys.stderr)
+        return False
+    tar.extractall(target_dir)
     tar.close()
     return True
 
 
+def download_platform(base_url, base_name, target_dir='.'):
+    # download base package
+    url = base_url + '/' + base_name + '.tar.gz'
+    status = download_and_extract(url, target_dir)
+    if not status:
+        return False
+    # download platform-specific package
+    url = base_url + '/' + base_name + '-' + PYNE_PLATFORM + '.tar.gz'
+    status = download_and_extract(url, target_dir)
+    if not status:
+        return False
+    return True
+
+
+DATA_URL = 'http://raw.githubusercontent.com/pyne/data/master'
+DECAY_H = os.path.join('src', 'decay.h')
+DECAY_CPP = os.path.join('src', 'decay.cpp')
+DECAY_H_REP = os.path.join('src', '_decay.h')
+DECAY_CPP_REP = os.path.join('src', '_decay.cpp')
+DECAY_URL = 'http://raw.githubusercontent.com/pyne/data/master/decay.tar.gz'
+
+
+def download_decay():
+    return download_platform(DATA_URL, 'decay', 'src')
+
+
 CRAM_H = os.path.join('src', 'decay.h')
```

Comment:
These need to be corrected to `cram.[hc]`

Reply:
In general, I expect the set of `cram` files to be similar to the set of `decay` files.